#!/usr/bin/env python3
PACKAGE = "line_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("initial_crop_top", int_t, 0, "Initial crop top", 50, 0, 100)
gen.add("initial_crop_bottom", int_t, 0, "Initial crop bottom", 100, 0, 100)
gen.add("initial_crop_left", int_t, 0, "Initial crop left", 0, 0, 100)
gen.add("initial_crop_right", int_t, 0, "Initial crop right", 100, 0, 100)

gen.add("use_live_crop", bool_t, 0, "Use live cropping", False)
gen.add("live_crop_weight", double_t, 0, "Live cropping weight", 0.5, 0.0, 1.0)

gen.add("use_median_blur", bool_t, 0, "Use median blur", True)

gen.add("filter_white", bool_t, 0, "Filter white", True)
gen.add("white_thresh", int_t, 0, "White threshold", 220, 0, 255)

gen.add("use_canny", bool_t, 0, "Use Canny edge detection", True)
gen.add("canny_lower_thresh", int_t, 0, "Canny lower threshold", 50, 0, 255)
gen.add("canny_upper_thresh", int_t, 0, "Canny upper threshold", 150, 0, 255)

gen.add("display_preprocessed_image", bool_t, 0, "Display preprocessed image", False)

exit(gen.generate(PACKAGE, "line_detection", "Preprocessor"))
